tests/test_wait_signal.py

```diff
@@ -312,3 +313,34 @@ def test_connected_signal(self, qtbot, signaller):
             blocker.connect(signaller.signal_args_2)
             signaller.signal_args_2.emit('foo', 2342)
         assert blocker.args == ['foo', 2342]
+
+
+class TestAssertNotEmitted:
+
+    """Tests for qtbot.assertNotEmitted."""
+
+    def test_not_emitted(self, qtbot, signaller):
+        with qtbot.assertNotEmitted(signaller.signal):
+            pass
+
+    def test_emitted(self, qtbot, signaller):
+        with pytest.raises(SignalEmittedError) as excinfo:
+            with qtbot.assertNotEmitted(signaller.signal):
+                signaller.signal.emit()
+
+        fnmatch.fnmatchcase(str(excinfo.value),
+                            "Signal * unexpectedly emitted.")
```

Comment:
Using `fnmatch` here because the `repr` of a signal is probably quite different between PyQt4/PyQt5/PySide.